### PR TITLE
Math Magicians React Project - Calculator Logic

### PR DESCRIPTION
**Pull request by jmontanholi**
## In this feature branch I implemented the following:

* ### Calculator logic code provided by Microverse. ✔️ 
* ### setState logic to use the calculator and display the results. ✔️

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,40 +1,60 @@
...
+      operation: '',
     };
   }
 
+  handleStateChange(newState) {
+    if (newState.total != null) {
+      this.setState({ total: newState.total });
+    }
+
+    // if (newState.next !== null) {
```
Comment (Dipeshtwis) [inline src/components/calculator.js]: - [x] I would recommend removing the commented code as removing commented code makes the codebase cleaner.

File: src/components/calculator.js [hunk 2]
```diff
@@ -1,40 +1,60 @@
...
+    this.setState({ next: newState.next });
+    // }
+
+    if (newState.operation != null) {
+      this.setState({ operation: newState.operation });
+    }
+  }
```
Comment (Dipeshtwis) [inline src/components/calculator.js:26]: - [x] After completeing one operation Ex: `1+2 = 3`, I tried to do another operation ` + 9`, but it shows me the error of `invalid number`. Please fix it for a smoother calculation.

The smoother calculation is:
```
1 + 2 = 3 then + 9 = 12
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @jmontanholi, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: You have implemented the click event for all buttons.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- The app is looking great and working fine, but when it comes to User experience, it's not that good. Currently, I am not able to see which operation I choose on the screen. Ex: If I try to do `4 + 5 `, then I won't able to see the `+` sign on the screen. Would you mind changing it up a bit (in terms of the display), a user to see the digits (and, hopefully, operations that get typed for great User experience?

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @jmontanholi  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**